src/femtomes.cpp

```diff
@@ -0,0 +1,534 @@
+	_correction_output_activated = false;
+	_configure_done = false;
+
+	/** Try different baudrates (115200 is the default for Femtomes) and request the baudrate that we want.	 */
+	const unsigned baudrates_to_try[] = {9600, 38400, 19200, 57600, 115200};
+	bool success = false;
+
+	unsigned test_baudrate;
+
+	for (unsigned int baud_i = 0; !success && baud_i < sizeof(baudrates_to_try) / sizeof(baudrates_to_try[0]); baud_i++) {
+		test_baudrate = baudrates_to_try[baud_i];
+
+		if (baudrate > 0 && baudrate != test_baudrate) {
+			continue; /**< skip to next baudrate*/
+		}
+
+		setBaudrate(test_baudrate);
+		
+		FEMTO_DEBUG("baudrate set to %i", test_baudrate);
+
+		for (int run = 0; run < 2; ++run) { /** try several times*/
+			if (writeAckedCommandFemto("UNLOGALL\r\n", "<UNLOGALL OK", FEMO_RESPONSE_TIMEOUT) == 0 && 
+				writeAckedCommandFemto("VERSION\r\n", "<VERSION OK", FEMO_RESPONSE_TIMEOUT) == 0) {
+				FEMTO_DEBUG("got port for baudrate %i", test_baudrate);
+				success = true;
+				break;
+			}
+		}
+	}
+	
+	if (!success) {
+		FEMTO_DEBUG("femtomes gps start failed %i", test_baudrate);
+		return -1;
+	}
+	/** 
+	* We successfully got a response and know to which port we are connected. Now set the desired baudrate
+	* if it's different from the current one.
+	*/
+	const unsigned desired_baudrate = 115200; /**< changing this requires also changing the SPD command*/
+
+	baudrate = test_baudrate;
+
+	if (baudrate != desired_baudrate) {
+		baudrate = desired_baudrate;
+		//@TODO
+		const char baud_config[] = "com%c 115200\r\n"; // configure baudrate to 115200
+		char baud_config_str[sizeof(baud_config)];
+		int len = snprintf(baud_config_str, sizeof(baud_config_str), baud_config, _port);
+		write(baud_config_str, len);
+		decodeInit();
+		receiveWait(200);
+		decodeInit();
+		setBaudrate(baudrate);
+
+		success = false;
+
+		for (int run = 0; run < 10; ++run) {
+			/** We ask for the port config again. If we get a reply, we know that the changed settings work.*/
+			if (writeAckedCommandFemto("UNLOGALL\r\n", "<UNLOGALL OK",FEMO_RESPONSE_TIMEOUT) == 0 && 
+				writeAckedCommandFemto("VERSION\r\n", "<VERSION OK",FEMO_RESPONSE_TIMEOUT) == 0) {
+				success = true;
+				break;
+			}
+		}
+
+		if (!success) {
+			return -1;
+		}
+
+		FEMTO_DEBUG("Successfully configured the baudrate");
+	}
+	PX4_INFO("Successfully configured the baudrate");
+#if 0
+	const char *config_options[][2] = {
+		"UNLOGALL\r\n",     "<UNLOGALL OK"    		/**< disable all NMEA and NMEA-Like Messages*/
+		"LOG UAVGPSB 0.2\r\n", "<LOG OK"   	/**< disable all ATM (ATOM) Messages*/
+	};
+
+	for (unsigned int conf_i = 0; conf_i < sizeof(config_options) / sizeof(config_options[0]); conf_i++) {
+		if (writeAckedCommandFemto(config_options[conf_i][0], config_options[conf_i][1], FEMO_RESPONSE_TIMEOUT) != 0) {
+			FEMTO_DEBUG("command %s failed", config_options[conf_i][0]);
+		}
+	}
+#endif
+	if (writeAckedCommandFemto("LOG UAVGPSB 0.2\r\n", "<LOG OK",FEMO_RESPONSE_TIMEOUT) == 0){

```

Comment:
Is that 5Hz? Can it go higher?

Reply:
123

Reply:
Yes you can use 20Hz.

Reply:
this have changed 5HZ to 20HZ in the code